kolibri/content/utils/transfer.py

```diff
@@ -141,7 +142,10 @@ def __init__(self, *args, **kwargs):
         super(FileDownload, self).__init__(*args, **kwargs)
 
     def start(self):
-        assert not self.started, "File download has already been started, and cannot be started again"
+        # If a file download was stopped by Internet connection error,
+        # then reopen the temp file to write to it
+        if self.started:
+            self.dest_file_obj = open(self.dest_tmp, "ab")
```

Comment:
how should I test that this is working as expected? Does it allow the server to be restarted in the middle of a transfer?

Reply:
Yes, to test it, I just created a new channel `03ff778aac984323a2a23216871022b5` with a large video file. Please turn off the wifi after the file `KOLIBRI_HOME/content/storage/b/f/bf6ce40da9e61bf2c133c3b1ceebf05d.mp4.transfer` exists in the filesystem (It takes about 10-20 seconds after the import starts). And then feel free to turn the wifi back on

Reply:
Hmm, you're appending, but might it be starting the download from the beginning (if the previous connected timed out, for example)? Have you checked that it doesn't end up with a larger than expected (and corrupt) file?

Reply:
Yes I checked that the file size is the same as the original from studio, and it's not corrupted.

Reply:
I've been playing with this for a while, and I have a few things I learned in the process.

Due to [this line](https://github.com/learningequality/kolibri/blob/develop/kolibri/content/utils/transfer.py#L141), we often download content in big chunks. On my computer, downloading the 90MB video file above doesn't ping the progress callback or write to disk at all until it's completed (if I comment out that line, I get granular progress reporting again). This made me worried that perhaps you didn't see corruption because it was starting from an empty file when "resuming". However, when I remove that line and have it use smaller chunks, I don't see corruption either. But when you interrupt the connection and let it time out, then reconnect, it seems to start downloading the file from the beginning anyway (the file size and progress bar both start back at 0). So I don't think this chunk of code is doing what you intended anyway.

I'd recommend we remove the "resume partial file" code here, as I don't think it's working, and then open an issue to address that (which I think can probably be done using the "byte-range" method I discussed in another comment). Instead, for now probably best to just cancel/close the file object and start a new one, if it times out.

Reply:
thank you for testing @jamalex ! I probably rebased before, but I saw all the comments are still in the `conversation` page. Some of them are hidden under `show outdated`. But thank you for the suggestion! I will not rebase so early in the future PRs.
I agree that I didn't see the corruption because the `.transfer` file was empty. 
Since the file object will be closed when it gets an exception and exits the `with` statement, I think I'm going to just change the open mode from `ab` to `wb` to start the new `.transfer` file object.